.ci/gitlab/scripts/common/write_dotenv.py

```diff
@@ -0,0 +1,50 @@
+import os
+import sys
+
+CI_PROJECT_DIR = "."
+CI_AUX_DIR = os.path.join(".ci", "tmp", "spack")
+
+
+def configure_packages_repo(dotenv):
+    dotenv["SPACK_PACKAGES_ROOT"] = CI_PROJECT_DIR
+    dotenv["SPACK_ROOT"] = CI_AUX_DIR
+
+
+def configure_core_repo(dotenv):
+    dotenv["SPACK_PACKAGES_ROOT"] = CI_AUX_DIR
+    dotenv["SPACK_ROOT"] = CI_PROJECT_DIR
+
+
+def read_dotenv(file):
+    dotenv = {}
+    for line in file:
+        if "=" in line:
+            key, value = line.rsplit("=", 1)
+            dotenv[key] = value.strip()
+    return dotenv
+
+
+def write_dotenv(env, file):
+    for var, value in env.items():
+        file.write(f"{var}={value}\n")
+
+
+if __name__ == "__main__":
+    dotenv = {}
+    base_dotenv = sys.argv[1] if len(sys.argv) > 1 else None
+    if os.path.exists(base_dotenv):
+        print(f"Found base dotenv: {base_dotenv}")
+        with open(base_dotenv, "r", encoding="utf-8") as fd:
+            dotenv = read_dotenv(fd)
+
+    repo_override = os.environ.get("CI_PROJECT_NAME")
+    if repo_override:
+        if repo_override.endswith("packages"):
+            configure_packages_repo(dotenv)
+        else:
+            configure_core_repo(dotenv)
+    else:
+        configure_packages_repo(dotenv)
```

Comment:
```suggestion
    if repo_override and not repo_override.endswith("packages"): 
        configure_core_repo(dotenv)
    else:
        configure_packages_repo(dotenv)
```
It might also be better to validate that `CI_PROJECT_NAME` is valid and use the full string instead of `endswith`

Reply:
I am not too overly concerned about this script being very robust right now since it is only going to be run by our CI. If there is interest by others to use it for some reason then I think it would be worth cleaning up and making more robust.

Reply:
The default of what happens here now is nothing is configured/the base dot env is written as a pass-through. I think that is the correct behavior.